src/main/java/com/flowingcode/vaadin/addons/gridexporter/ExcelStreamResourceWriter.java

```diff
@@ -411,39 +411,47 @@ private Cell findCellWithPlaceHolder(Sheet sheet, String placeholder) {
     return null;
   }
 
-  private void fillHeaderOrFooter(
-      Sheet sheet,
-      Cell headersOrFootersCell,
-      List<Pair<String, Column<T>>> headersOrFooters,
-      boolean isHeader) {
-    CellStyle style = headersOrFootersCell.getCellStyle();
-    sheet.setActiveCell(headersOrFootersCell.getAddress());
-    headersOrFooters.forEach(
-        headerOrFooter -> {
-          if (!isHeader) {
-            // clear the styles before processing the column in the footer
-            ComponentUtil.setData(headerOrFooter.getRight(), COLUMN_CELLSTYLE_MAP, null);
-          }
-          Cell cell =
-              sheet
-                  .getRow(sheet.getActiveCell().getRow())
-                  .getCell(sheet.getActiveCell().getColumn());
-          if (cell == null) {
-            cell =
-                sheet
-                    .getRow(sheet.getActiveCell().getRow())
-                    .createCell(sheet.getActiveCell().getColumn());
-          }
-          cell.setCellStyle(style);
-          Object value =
-              (isHeader
-                  ? headerOrFooter.getLeft()
-                  : transformToType(headerOrFooter.getLeft(), headerOrFooter.getRight()));
-          buildCell(value, cell, headerOrFooter.getRight(), null);
-          configureAlignment(headerOrFooter.getRight(), cell, isHeader?ExcelCellType.HEADER:ExcelCellType.FOOTER);
-          sheet.setActiveCell(
-              new CellAddress(
-                  sheet.getActiveCell().getRow(), sheet.getActiveCell().getColumn() + 1));
-        });
+  private void fillFooter(Sheet sheet, Cell headersOrFootersCell,
+      List<Pair<String, Column<T>>> headersOrFooters, boolean isHeader) {
+    List<Pair<List<String>, Column<T>>> headersOrFootersCellSingleRow = headersOrFooters.stream()
+        .map(pair -> Pair.of(List.of(pair.getLeft()), pair.getRight())).collect(Collectors.toList());
+    fillHeaderOrFooter(sheet, headersOrFootersCell, headersOrFootersCellSingleRow, isHeader);
   }
+  private void fillHeaderOrFooter(Sheet sheet, Cell headersOrFootersCell,
+      List<Pair<List<String>, Column<T>>> headersOrFooters, boolean isHeader) {
+    CellStyle style = headersOrFootersCell.getCellStyle();
+    
+    int startRow = headersOrFootersCell.getRowIndex();
+    int currentColumn = headersOrFootersCell.getColumnIndex();
+    boolean shiftFirstTime = true;
+    for (Pair<List<String>, Column<T>> headerOrFooter : headersOrFooters) {
+      List<String> headerOrFooterTexts = headerOrFooter.getLeft();
+      Column<T> column = headerOrFooter.getRight();
```

Comment:
I analyzed the code and how it is today it is impossible to have those variables to be null. Another thing is that the exception proposed to be thrown would cause confusion. I prefer to avoid doing this modification.